packages/@aws-cdk/aws-servicecatalogappregistry/lib/attribute-group.ts

```diff
@@ -0,0 +1,113 @@
+import * as cdk from '@aws-cdk/core';
+import { InputValidator } from './private/validation';
+import { CfnAttributeGroup } from './servicecatalogappregistry.generated';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from 'constructs';
+
+/**
+ * A Service Catalog AppRegistry Attribute Group.
+ */
+export interface IAttributeGroup extends cdk.IResource {
+  /**
+   * The ARN of the attribute group.
+   * @attribute
+   */
+  readonly attributeGroupArn: string;
+
+  /**
+   * The ID of the attribute group.
+   * @attribute
+   */
+  readonly attributeGroupId: string;
+}
+
+/**
+ * Attributes for the attribute group, defined by a nested list of
+ * key-value pairs.
+ */
+export interface Attribute {
+  [key: string]: string | number | boolean | Attribute |
+  string[] | number[] | boolean[] | Attribute[];
+}
+
+/**
+ * Properties for a Service Catalog AppRegistry Attribute Group
+ */
+export interface AttributeGroupProps {
+  /**
+   * Enforces a particular physical attribute group name.
+   */
+  readonly attributeGroupName: string;
+
+  /**
+   * Description for attribute group.
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * A JSON of nested key-value pairs that represent the attributes in the group.
+   * Attributes maybe an empty JSON '{}', but must be explicitly stated.
+   */
+  readonly attributes: Attribute;
```

Comment:
> One thing that will probably raise eyebrows is the attribute field. We have explicit guidance to make this field required, even though an empty json is valid `{}`. The reason for this is that we don't want end users to forget about this field when creating attribute group, since it's basically the main item of the resource. It is possible to have workflows where defining it without attributes first is done.

Not sure I follow, sorry. If it's valid to have this empty, then it should not be required.

Reply:
Hence why I proactively tried to address this, since yes it does go against the convention of cdk.  I reached out to the dev team for comment and the gist again was basically that making the field optional would make it easier for customer to make a mistake of forgetting to populate an optional field when they would almost always want to.

Reply:
It is a rare but valid use case for it to be empty, the entire construct is basically a wrapper around this field.

Reply:
OK. Let's leave it as-is then.